lib/linters/example-linter.js

```diff
@@ -94,7 +94,8 @@ module.exports = async function (argv, tap) {
 
     const configs = []
 
-    steps.forEach((step) => {
+    // Flatten group steps before processing
+    steps.flatMap((step) => step.group && step.steps || step).forEach((step) => {
```

Comment:
Would you mind moving the `flatMap` to before the `configs` variable is defined with a comment (similar to the one in line 93) so it is clear that if the step is a group step, we want to validate those nested inside of it.

Reply:
@toote 
Sure, moved the flattening out with a comment